docs/resources/environment.md

```diff
@@ -0,0 +1,153 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "rafay_environment Resource - terraform-provider-rafay"
+subcategory: ""
+description: |-
+  
+---
+
+# rafay_environment (Resource)
+
+
+
+## Example Usage
+
+```terraform
+resource "rafay_environment" "eks-rds-env-example" {
+  metadata {
+    name    = var.name
+    project = var.project
+  }
+  spec {
+    template {
+      name    = var.et_name
+      version = var.et_version
+    }
+    variables {
+      name       = "name"
+      value_type = "text"
+      value      = "dev-env-resource"
+      options {
+        description = "this is the name of resource created"
+        sensitive   = false
+        required    = true
+      }
+    }
+  }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `metadata` (Block List, Max: 1) Metadata of the environment resource (see [below for nested schema](#nestedblock--metadata))
+- `spec` (Block List, Max: 1) Specification of the environment resource (see [below for nested schema](#nestedblock--spec))
+
+### Optional
+
+- `timeouts` (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
+
+<a id="nestedblock--metadata"></a>
+### Nested Schema for `metadata`
+
+***Required***
+
+- `name` (String) name of the resource
+- `project` (String) Project of the resource
+
+***Optional***
+
+- `description` (String) description of the resource
+
+
+<a id="nestedblock--spec"></a>
+### Nested Schema for `spec`
+
+***Required***
+
+- `template` (Block List, Max: 1) Environment resource template (see [below for nested schema](#nestedblock--spec--template))
+
+***Optional***
+
+- `agents` (Block List) Agents that are eligible to process the environment (see [below for nested schema](#nestedblock--spec--agents))
+- `sharing` (Block List, Max: 1) Defines if this is shared with other projects (see [below for nested schema](#nestedblock--spec--sharing))
+- `variables` (Block List) Variables data for environment to be created (see [below for nested schema](#nestedblock--spec--variables))
+
+<a id="nestedblock--spec--agents"></a>
+### Nested Schema for `spec.agents`
+
+***Required***
+
+- `name` (String) name of the resource
+
+
+<a id="nestedblock--spec--sharing"></a>
+### Nested Schema for `spec.sharing`
+
+***Optional***
+
+- `enabled` (Boolean) flag to specify if sharing is enabled for resource
+- `projects` (Block List) list of projects this resource is shared to (see [below for nested schema](#nestedblock--spec--sharing--projects))
+
+<a id="nestedblock--spec--sharing--projects"></a>
+### Nested Schema for `spec.sharing.projects`
+
+***Optional***
+
+- `name` (String) name of the project
+
+
+
+<a id="nestedblock--spec--template"></a>
+### Nested Schema for `spec.template`
+
+***Required***
+
+- `name` (String) name of the resource
```

Comment:
change "name of the resource" to "name of the environment template"
change "version of the resource" to "version of the environment template"